codec-http2/src/test/java/io/netty/handler/codec/http2/Http2ConnectionHandlerTest.java

```diff
@@ -243,6 +245,17 @@ public void serverReceivingInvalidClientPrefaceStringShouldHandleException() thr
     }
 
     @Test
+    public void serverReceivingHttp1ClientPrefaceStringShouldIncludePreface() throws Exception {
+        when(connection.isServer()).thenReturn(true);
+        handler = newHandler();
+        handler.channelRead(ctx, copiedBuffer("GET /path HTTP/1.1", US_ASCII));
+        ArgumentCaptor<ByteBuf> captor = ArgumentCaptor.forClass(ByteBuf.class);
+        verify(frameWriter).writeGoAway(eq(ctx), eq(0), eq(PROTOCOL_ERROR.code()),
```

Comment:
The `frameWriter` mock is configured in `setup` to release the buffer so each test doesn't have to worry about this (see below). For this test since we want to verify the content we should install a new handler for `writeGoAway` so we can verify the content of the message.

```java
doAnswer(new Answer<ChannelFuture>() {
            @Override
            public ChannelFuture answer(InvocationOnMock invocation) throws Throwable {
                ByteBuf buf = invocation.getArgumentAt(3, ByteBuf.class);
                buf.release();
                return future;
            }
        }).when(frameWriter).writeGoAway(
                any(ChannelHandlerContext.class), anyInt(), anyInt(), any(ByteBuf.class), any(ChannelPromise.class));
```